vm/src/obj/objset.rs

```diff
@@ -813,7 +818,68 @@ impl TryFromObject for SetIterable {
     }
 }
 
+#[derive(Copy, Clone, Default)]
+struct SetSizeInfo {
+    size: Option<usize>,
+    position: usize,
+}
+
+#[pyclass]
+struct PySetIterator {
+    dict: PyRc<SetContentType>,
+    size_info: crossbeam_utils::atomic::AtomicCell<SetSizeInfo>,
+}
+
+impl fmt::Debug for PySetIterator {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // TODO: implement more detailed, non-recursive Debug formatter
+        f.write_str("setiterator")
+    }
+}
+
+#[pyimpl]
+impl PySetIterator {
+    #[pymethod(name = "__next__")]
```

Comment:
`#[pymethod(magic)]` also works.
When funciton name is `f`, `magic` attribute turns the python attribute name into `__f__`.

Reply:
Thanks, that looks great. I changed them to `#[pymethod(magic)]`.